docroot/modules/humsci/hs_migrate/src/Form/HsMigrateImportForm.php

```diff
@@ -0,0 +1,220 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(MigrationPluginManagerInterface $migrations_manager, DateFormatterInterface $date_formatter, KeyValueFactoryInterface $key_value, AccountProxyInterface $account) {
+    $this->dateFormatter = $date_formatter;
+    $this->lastMigrations = $key_value->get('migrate_last_imported');
+    $this->migrationManager = $migrations_manager;
+    $this->account = $account;
+
+    $this->migrations = $this->migrationManager->createInstances([]);
+
+    // No need to show migrations that are dependencies. They will get executed
+    // when their dependent migration is executed.
+    foreach ($this->migrations as $migration) {
+      foreach ($migration->getMigrationDependencies()['required'] as $dependency) {
+        // Set the dependency migrations to idle so that the primary migration
+        // will still execute.
+        $this->migrations[$dependency]->interruptMigration(MigrationInterface::RESULT_STOPPED);
+        $this->migrations[$dependency]->setStatus(MigrationInterface::STATUS_IDLE);
+        unset($this->migrations[$dependency]);
+      }
+    }
+
+    // Remove migrations that the user doesn't have access to.
+    foreach (array_keys($this->migrations) as $migration_id) {
+      if (!$this->account->hasPermission("import $migration_id migration")) {
+        unset($this->migrations[$migration_id]);
+      }
+    }
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'hs_migrate_import_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = [];
+    $form['table'] = [
+      '#type' => 'table',
+      '#header' => $this->buildHeader(),
+      '#empty' => $this->t('No migrations found'),
+    ];
+    foreach ($this->migrations as $migration_id => $migration) {
+      $form['table'][$migration_id] = $this->buildRow($migration);
+    }
+
+    return $form;
+  }
+
+  /**
+   * Build the table header labels.
+   *
+   * @return array
+   *   Array of table headers.
+   */
+  protected function buildHeader() {
+    return [
+      $this->t('Importer'),
+      $this->t('Status'),
+      $this->t('Imported Items'),
+      $this->t('Last Imported'),
+      $this->t('Import'),
+    ];
+  }
+
+  /**
+   * Build the form row for the given migration object.
+   *
+   * @param \Drupal\migrate\Plugin\MigrationInterface $migration
+   *   Migration plugin object.
+   *
+   * @return array
+   *   Form render array.
+   */
+  protected function buildRow(MigrationInterface $migration) {
+    $row['label']['#markup'] = $migration->label();
+    $row['status']['#markup'] = $migration->getStatusLabel();
+    $row['imported']['#markup'] = $migration->getIdMap()->importedCount();
+
+    if ($last_imported = $this->lastMigrations->get($migration->id(), FALSE)) {
+      $row['last_imported']['#markup'] = $this->dateFormatter->format($last_imported / 1000, 'custom', 'M j Y g:i a');
```

Comment:
What format is $last_imported in and why `/1000`?

Reply:
microtime. this is an excerpt from migrate_tools module
```
    $migrate_last_imported_store = \Drupal::keyValue('migrate_last_imported');
    $migrate_last_imported_store->set($event->getMigration()->id(), round(microtime(TRUE) * 1000));
```